File: frontend/src/types/payment.ts
export interface Payment {
  id: string;
  payment_number: string;
  invoice_id: string;
  amount: string;
  payment_date: string;
  payment_method: PaymentMethod;
  transaction_reference: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface PaymentListItem {
  id: string;
  payment_number: string;
  invoice_id: string;
  amount: string;
  payment_date: string;
  payment_method: PaymentMethod;
  transaction_reference: string | null;
  created_at: string;
}

export type PaymentMethod = "CASH" | "UPI" | "BANK_TRANSFER" | "CHEQUE";

export const PAYMENT_METHOD_LABELS: Record<PaymentMethod, string> = {
  CASH: "Cash",
  UPI: "UPI",
  BANK_TRANSFER: "Bank Transfer",
  CHEQUE: "Cheque",
};

export const PAYMENT_METHOD_COLORS: Record<PaymentMethod, string> = {
  CASH: "bg-green-100 text-green-700",
  UPI: "bg-blue-100 text-blue-700",
  BANK_TRANSFER: "bg-indigo-100 text-indigo-700",
  CHEQUE: "bg-purple-100 text-purple-700",
};

export interface PaymentListResponse {
  items: PaymentListItem[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

File: backend/app/schemas/payment.py
"""Pydantic v2 schemas for the Payment module."""

from __future__ import annotations

from datetime import date, datetime
from decimal import Decimal
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class PaymentCreate(BaseModel):
    invoice_id: UUID
    amount: Decimal = Field(..., gt=0, description="Must be greater than zero")
    payment_date: date
    payment_method: str = "CASH"
    transaction_reference: Optional[str] = None
    notes: Optional[str] = None


class PaymentOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    payment_number: str
    invoice_id: UUID
    amount: Decimal
    payment_date: date
    payment_method: str
    transaction_reference: Optional[str] = None
    notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class PaymentListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    payment_number: str
    invoice_id: UUID
    amount: Decimal
    payment_date: date
    payment_method: str
    transaction_reference: Optional[str] = None
    created_at: datetime


class PaymentListResponse(BaseModel):
    items: list[PaymentListItem]
    total: int
    page: int
    page_size: int
    total_pages: int
